crates/rpc/rpc/src/eth/sim_bundle.rs

```diff
@@ -21,20 +72,409 @@ impl<Eth> EthSimBundle<Eth> {
+                            EthSimBundleError::InvalidBundle.to_string(),
+                        ));
+                    }
+                }
+            }
+        }
+
+        Ok(items)
+    }
+
+    async fn sim_bundle(
         &self,
         request: SendBundleRequest,
         overrides: SimBundleOverrides,
-    ) -> RpcResult<SimBundleResponse> {
-        info!("mev_simBundle called, request: {:?}, overrides: {:?}", request, overrides);
-        Err(EthApiError::Unsupported("mev_simBundle is not supported").into())
+        logs: bool,
+    ) -> Result<SimBundleResponse, Eth::Error> {
+        let SimBundleOverrides {
+            parent_block,
+            block_number,
+            coinbase,
+            timestamp,
+            gas_limit,
+            base_fee,
+            ..
+        } = overrides;
+
+        // Parse and validate bundle
+        // Also, flatten the bundle here so that its easier to process
+        let flattened_bundle = self.parse_and_flatten_bundle(&request)?;
+
+        let block_id = parent_block.unwrap_or(BlockId::Number(BlockNumberOrTag::Latest));
+        let (cfg, mut block_env, current_block) = self.eth_api().evm_env_at(block_id).await?;
+
+        let parent_header = RpcNodeCore::provider(&self.inner.eth_api)
+            .header_by_number(block_env.number.saturating_to::<u64>())
+            .map_err(EthApiError::from_eth_err)? // Explicitly map the error
+            .ok_or_else(|| {
+                EthApiError::HeaderNotFound((block_env.number.saturating_to::<u64>()).into())
+            })?;
+
+        // apply overrides
+        if let Some(block_number) = block_number {
+            block_env.number = U256::from(block_number);
+        }
+
+        if let Some(coinbase) = coinbase {
+            block_env.coinbase = coinbase;
+        }
+
+        if let Some(timestamp) = timestamp {
+            block_env.timestamp = U256::from(timestamp);
+        }
+
+        if let Some(gas_limit) = gas_limit {
+            block_env.gas_limit = U256::from(gas_limit);
+        }
+
+        if let Some(base_fee) = base_fee {
+            block_env.basefee = U256::from(base_fee);
+        } else if cfg.handler_cfg.spec_id.is_enabled_in(SpecId::LONDON) {
+            if let Some(base_fee) = parent_header.next_block_base_fee(
+                RpcNodeCore::provider(&self.inner.eth_api)
+                    .chain_spec()
+                    .base_fee_params_at_block(block_env.number.saturating_to::<u64>()),
+            ) {
+                block_env.basefee = U256::from(base_fee);
+            }
+        }
+
+        if cfg.handler_cfg.spec_id.is_enabled_in(SpecId::CANCUN) {
+            let excess_blob_gas = calc_excess_blob_gas(
+                parent_header.excess_blob_gas.unwrap_or_default(),
+                parent_header.blob_gas_used.unwrap_or_default(),
+            );
+
+            block_env.set_blob_excess_gas_and_price(excess_blob_gas);
+        }
```

Comment:
you're right, `excess_blob_gas` doesn't need to be set here, but `base_fee` is an override so we should keep it

Reply:
golang impl uses latest not pending, but I dont see a reason why not use pending to simulate bundles: https://github.com/flashbots/builder/blob/df9c765067d57ab4b2d0ad39dbb156cbe4965778/internal/ethapi/sbundle_api.go#L227

Reply:
basically we need to ensure that block env fields correspond to block latest + 1

I think this can be done by just using pending block for header, and latest block for env